src/components/Button/Button.tsx

```diff
@@ -67,13 +82,14 @@ const Button: React.FC<Props & TestProps & EventProps> = props => {
         iconRight,
         childrenCount: React.Children.count(children),
       })}
-      onClick={onClick}
+      onClick={handleAsyncOperation}
```

Comment:
I am so familiar with the codebase, so maybe i miss something. If there there is a `submit` button inside a form for example, so the form should be submitted with enter key or with submit event, right? If so, how we can set the button in the loading state?

Reply:
At the moment, our buttons do not have a specific type, which semantically incorrect. Although, based on the buttons' usage is not required for now.
I already planned to open a bug in the DS board. We can introduce a new prop buttonType ('submit', 'reset', 'button'). 

Then we can investigate accessing the loading state using keys or button types.

Reply:
ok! thank you for the reply!